pycbc/pnutils.py

```diff
@@ -530,6 +533,8 @@ def _get_imr_duration(m1, m2, s1z, s2z, f_low, approximant="SEOBNRv4"):
         )
     else:
         raise RuntimeError("I can't calculate a duration for %s" % approximant)
+
+    assert time_length > 0, f"Got a negative IMR duration of {time_length} for {approximant} with params (m1, m2, s1z, s2z, f_low) = ({m1}, {m2}, {s1z}, {s2z}, {f_low})"
```

Comment:
I'm think that this should be a RuntimeError, though we haven't been treating it as an error before. 

It may be best to ask in #pycbc-code slack whether there are any use cases where `_get_imr_duration` is allowed to be negative before adding this

Reply:
Assertion is similar in its effect to RuntimeError afaik .. I'm not aware of any good use case for negative durations, maybe the best way of checking everything possible is to see if putting in this assertion or error does cause anything else to break.

Reply:
The assertion does work as an instance of the `Assertion Error`. I think this is an appropriate use case for this.
I am trying to run a large mass ratio (upto 1000) search with IMRPhenomXAS. When I tried and used PhenomD duration as a test, I got a few negative durations. Hence the assertion.

Reply:
The Phenom duration code is known to produce negative durations. The question is do you want the code to fail in that case (in that case raise an appropriate Error ... I don't think we've really got much stuff doing error handling here, so as long as it's documented AssertionError seems fine)? OR do you want the code to realise that it's just the duration function that's broken and you can still get valid waveforms (hopefully valid waveforms) in some cases when the duration is reported as negative.

Reply:
I want to use it to perform SSM and usual BBH searches up to mass ratio 1000. So duration is used in the statistics. 
For now, I have removed templates with a negative duration. So the assert statement in there just for the safety.

Should I remove it and add it as `logging.warn` ?

Reply:
well, so maybe we leave it up to the code that uses the duration function to check for negative-ness rather than _always_ error.  In that case we should probably leave a comment on the warning like 'Phenom duration code is known to produce negative durations [because of bla bla .. if we know why?], this does not necessarily indicate a problem with waveform generation'